javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -0,0 +1,486 @@
+
+  private val chars = ('A' to 'Z') ++ ('a' to 'z') ++ ('0' to '9')
+  private val basicEncodeTable: Array[Byte] = (chars ++ Seq('+', '/')).map(_.toByte).toArray
+  private val urlSafeEncodeTable: Array[Byte] = (chars ++ Seq('-', '_')).map(_.toByte).toArray
+
+  private def decodeTable(encode: Array[Byte]): Array[Int] = {
+    val decode: Array[Int] = Array.fill(256)(-1)
+    encode.zipWithIndex.foreach{ case (b,i) => decode(b) = i}
+    decode('=') = -2
+    decode
+  }
+
+  private val basicDecodeTable: Array[Int] = decodeTable(basicEncodeTable)
+  private val urlSafeDecodeTable: Array[Int] = decodeTable(urlSafeEncodeTable)
+
+  private val mimeLineSeparators = Array[Byte]('\r','\n')
+  private final val mimeLineLength = 76
+
+  private val basicEncoder = new Encoder(basicEncodeTable)
+  private val basicDecoder = new Decoder(basicDecodeTable, ignoreInvalid = false)
+
+  private val mimeEncoder = new Encoder(basicEncodeTable, mimeLineLength, mimeLineSeparators)
+  private val mimeDecoder = new Decoder(basicDecodeTable, ignoreInvalid = true)
+
+  private val urlSafeEncoder = new Encoder(urlSafeEncodeTable)
+  private val urlSafeDecoder = new Decoder(urlSafeDecodeTable, ignoreInvalid = false)
+
+  // --------------------------------------------------------------------------
+
+  def getEncoder(): Encoder = basicEncoder
+
+  def getUrlEncoder(): Encoder = urlSafeEncoder
+
+  def getMimeEncoder(): Encoder = mimeEncoder
+
+  def getMimeEncoder(lineLength: Int, lineSeparator: Array[Byte]): Encoder = {
+    lineSeparator.foreach { b =>
+      if (basicDecodeTable(b & 0xff) != -1)
+        throw new IllegalArgumentException("Illegal base64 line separator character 0x" + Integer.toString(b, 16))
+    }
+    new Encoder(basicEncodeTable, lineLength / 4 * 4, lineSeparator)
+  }
+
+  def getDecoder(): Decoder = basicDecoder
+
+  def getUrlDecoder(): Decoder = urlSafeDecoder
+
+  def getMimeDecoder(): Decoder = mimeDecoder
+
+  // --------------------------------------------------------------------------
+
+  class Decoder private[Base64](table: Array[Int], ignoreInvalid: Boolean) {
+
+    def decode(src: Array[Byte]): Array[Byte] = {
+      val dst = new Array[Byte](dstRequiredLength(src, 0, src.length))
+      doDecode(new Wrapper(src), new Wrapper(dst))
+      dst
+    }
+
+    def decode(src: String): Array[Byte] = decode(src.getBytes(StandardCharsets.ISO_8859_1))
+
+    def decode(src: Array[Byte], dst: Array[Byte]): Int = {
+      if (dst.length >= dstMaxLength(src.length)) // dst is definitely large enough
+        doDecode(new Wrapper(src), new Wrapper(dst))
+      else if (dst.length < dstRequiredLength(src, 0, src.length))
+        throw new IllegalArgumentException("Output byte array is too small for decoding all input bytes")
+      else
+        doDecode(new Wrapper(src), new Wrapper(dst))
+    }
+
+    def decode(buffer: ByteBuffer): ByteBuffer = {
+      val start = buffer.position()
+      try {
+        val src = buffer.array()
```

Comment:
`TypedArrayByteBuffer` is an implementation of `ByteBuffer` that is direct, and is not backed by an array, so `hasArray` returns false.

The implementation of `nio` buffers in Scala.js is highly typeclass-based. I'm not surprised that IntelliJ has trouble with it.